server.go

```diff
@@ -1395,9 +1398,16 @@ func (s *Server) Close() error {
 // closeListenerClients closes all clients on the specified listener.
 func (s *Server) closeListenerClients(listener string) {
 	clients := s.Clients.GetByListener(listener)
+	var wg sync.WaitGroup
 	for _, cl := range clients {
-		_ = s.DisconnectClient(cl, packets.ErrServerShuttingDown)
-	}
+		wg.Add(1)
+		go func(c *Client) {
```

Comment:
We may need to put some concurrent limits here, on a large server, this can be in the hundreds of thousands or even millions.

Reply:
Handling thousands of connections that need to be disconnected raises the question of how many Goroutines to limit. If the number of Goroutines is too low, it can lead to a situation where, after pressing Ctrl+C, the server takes a very long time to wait for all connections to disconnect.

Reply:
Few thousands (even tens of thousands) of Go routines should be fine. After all, for every connections, there are already 2 Go routines. It's just a matter how quickly these Go routines are spun up...  You can use [climit](https://github.com/thedevop1/climit) if you like.

Reply:
Thanks, [climit](https://github.com/thedevop1/climit) looks very nice. I'll give it a try later.

Reply:
@thedevop  I'm sorry for not using climit. climit is excellent, but the reason I didn't use it is that, when using climit, it didn't bring a significant improvement in code simplicity(there is also code for handling timeouts here).Instead, it concealed some details within climit. I extracted inspiration from climit, so thank you once again.